### PR TITLE
Bluetooth: lc3: Improve BT_CODEC_LC3_CHAN_COUNT_SUPPORT macro

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This improves the macro so that it can take variable number of arguments, e.g. BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 2, 8).

### CODE REVIEW

**Comment by Thalley** [comment]
What is the advantage of doing `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 3, 7)` compared to 
`BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(3) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(7)`?

Shorter lines of code?

**Comment by MariuszSkamra** [comment]
> What is the advantage of doing `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 3, 7)` compared to `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(3) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(7)`?
> 
> Shorter lines of code?

Yes :D

**Comment by Thalley** [comment]
> > What is the advantage of doing `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 3, 7)` compared to `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(3) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(7)`?
> > Shorter lines of code?
> 
> Yes :D

What do you think about the proposal in https://github.com/zephyrproject-rtos/zephyr/pull/45373#issuecomment-1118360118? 

While the variable-argument approach using the fixed size macros probably works, it's not particularly readable and easy to understand from a user's perspective if they investigate the macro.

**Comment by MariuszSkamra** [comment]
> > > What is the advantage of doing `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 3, 7)` compared to `BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(3) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(7)`?
> > > Shorter lines of code?
> > 
> > 
> > Yes :D
> 
> What do you think about the proposal in [#45373 (comment)](https://github.com/zephyrproject-rtos/zephyr/pull/45373#issuecomment-1118360118)?
> 
> While the variable-argument approach using the fixed size macros probably works, it's not particularly readable and easy to understand from a user's perspective if they investigate the macro.

You have right, it was not a bit ugly solution. Please check the latest PR

File: include/zephyr/bluetooth/audio/lc3.h [hunk 1]
```diff
@@ -129,22 +129,28 @@ enum bt_codec_capability_type {
...
+#define BT_CODEC_LC3_CHAN_COUNT_MAX 8
 /** @def BT_CODEC_LC3_CHAN_COUNT_SUPPORT
  *  @brief LC3 channel count support capability
- *  OR multiple supported counts together from 1 to 8
- *  Example to support 1 and 3 channels:
- *  BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1) | BT_CODEC_LC3_CHAN_COUNT_SUPPORT(3)
+ *
+ *  Macro accepts variable number of channel counts.
+ *  The allowed channel counts are defined by specification and have to be in range from
+ *  @ref BT_CODEC_LC3_CHAN_COUNT_MIN to @ref BT_CODEC_LC3_CHAN_COUNT_MAX.
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/lc3.h:145]:
```suggestion
 *  @ref BT_CODEC_LC3_CHAN_COUNT_MIN to @ref BT_CODEC_LC3_CHAN_COUNT_MAX inclusive.
```

File: include/zephyr/bluetooth/audio/lc3.h [hunk 2]
```diff
@@ -129,22 +129,28 @@ enum bt_codec_capability_type {
...
  */
-#define BT_CODEC_LC3_CHAN_COUNT_SUPPORT(_count)  ((uint8_t)BIT((_count) - 1))
-
+#define BT_CODEC_LC3_CHAN_COUNT_SUPPORT(...) ((uint8_t)((FOR_EACH(BIT, (|), __VA_ARGS__)) >> 1))
```
Comment (Thalley) [inline include/zephyr/bluetooth/audio/lc3.h:147]: Can you also provide an example here, like it had before? 
e.g.

```c
 * Example to support 1 and 3 channels:
 *  BT_CODEC_LC3_CHAN_COUNT_SUPPORT(1, 3)
```

**Review by Thalley — CHANGES_REQUESTED** [review]
New solution is much cleaner, thanks! 

Added a few requests

**Review by Thalley — DISMISSED** [review]

**Review by Thalley — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**